pkg/operator/ceph/cluster/mon/spec.go

```diff
@@ -330,7 +330,6 @@ func UpdateCephDeploymentAndWait(context *clusterd.Context, deployment *apps.Dep
 
 	callback := func(action string) error {
 		if !isUpgrade {
-			logger.Info("this is not a Ceph upgrade, not performing upgrade checks")
```

Comment:
Why not keep this message? If the Ceph image didn't change it's still valid, right?

Reply:
Because now it's polluting the logs a bit more since we just allowed a new condition. Now the only time when it'd not run is when someone tries to downscale which will fail anyway.
If the image didn't change we would never reach that code AFAIR or if that's the case we should not.

Reply:
Ok. To avoid a merge conflict with this PR, let's backport #4360 before this is backported.